Third parties 2021 markdown

tunetheweb opened the pull request:
Closes #2145 

Keeping in draft as awaiting Reviewer feedback:
- [x] - From @patrickhulce 
- [ ] - From @housseindjirdeh 
- [x] - From @csswizardry
- [x] - From @andydavies 
- [x] - From @simonhearne 

Staged version at: https://20211103t203721-dot-webalmanac.uk.r.appspot.com/en/2021/third-parties

rviscomi commented on src/content/en/2021/third-parties.md:
cc @maudnals 

```suggestion
Of course, one of the major reasons for using third parties on your site is for tracking for advertisement purposes, which by its very nature is not going to be in the best privacy interests of your visitors. Alternatives to this pervasive tracking, which is basically only possible by the use of third parties, <a hreflang="en" href="https://web.dev/floc/">have been suggested (FLoC)</a> but have failed to gain traction.
```

tunetheweb replied:
Changed it to this:

> Alternatives to this pervasive tracking, which is basically only possible by the use of third parties, have been suggested <a hreflang="en" href="https://blog.google/products/ads-commerce/2021-01-privacy-sandbox/">such as Google's Privacy Sandbox and FLoC iniative</a> but have failed to gain traction.

rviscomi commented on src/content/en/2021/third-parties.md:
```suggestion
What is perhaps more concerning is the tracking that can occur without website users and owners being aware. There is the old adage that if you're not paying for a product or service, then you are the product. Many third parties give away their product for "free", which for most means they are monetizing it in some other way—usually at the expense of your visitors' privacy!
```

rviscomi commented on src/content/en/2021/third-parties.md:
```suggestion
Adoption of newer technologies like <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy">`feature-policy` and `permission-policy`</a> can restrict the usage of certain functionalities of the browser, such as microphones and video cameras. These can reduce the privacy and security risks though many of these will usually be secured behind a browser prompt to ensure they are not silently activated. Google is also working on a <a hreflang="en" href="https://github.com/bslassey/privacy-budget">Privacy Budget proposal</a> to limit the privacy impact of web browser, though others <a hreflang="en" href="https://blog.mozilla.org/en/mozilla/google-privacy-budget-analysis/">remain skeptical of their work in this space</a>. All in all, adding privacy controls seems to be swimming against the tide given the intent of many third party resources.
```

rviscomi commented on src/content/en/2021/third-parties.md:
How can we make this section more data-driven?

tunetheweb replied:
I spent a good bit of time on this, honestly I did, trying to come up with something more data driven, but unfortunately did not. It's apparently the last two year's chapters had the same problem.

None-the-less, I think they are important topics to cover, and think the chapter would be incomplete without this section.

rviscomi commented on src/content/en/2021/third-parties.md:
```suggestion
Third parties are integral to the web. In many ways they are the web; without the prevalence of third parties, websites would be harder to build and less feature rich. As mentioned at the beginning, interconnectedness is at the very heart of the web, and third parties are the natural extension of this. Our analysis has shown that third parties are more prevalent than ever—sites without them are very much the exception!
```

rviscomi commented on src/content/en/2021/third-parties.md:
```suggestion
It's easy to get sucked into the negative however, so to finish off the chapter, let's look back at the positives. There is a reason that third parties are so prevalent and they are (usually!) used out of choice. Sharing is what the web is about and so third parties are very much in the spirit of the web. It's amazing what functionality we web developers have at our disposal and how easy it is to add them to our sites. Hopefully this chapter has opened your eyes to give a little more thought to making sure you fully understand the deal you're making when you do that.
```

rviscomi reviewed (changes requested):
I'm realizing now after having finished the review that I should have just opened a new PR with my edits rather than suggesting them all here 😅

tunetheweb commented on src/content/en/2021/third-parties.md:
This was a convention followed in the last two years, and kind of matches the quotes in the previous sentence. Does that change your opinion here?

rviscomi replied:
If it's a convention then SGTM

tunetheweb commented on src/content/en/2021/third-parties.md:
I've switched the the official definitions from here which are more consistent: https://github.com/patrickhulce/third-party-web/blob/master/data/categories.json. I've edited some of them down to first sentence.

tunetheweb commented on src/content/en/2021/third-parties.md:
But how can I do that within one graph and keep the label consistent?

And not entirely convinced about your graph. Looks a bit grey scale and difficult to differentiate between then.

rviscomi replied:
It would probably require one chart for desktop and one for mobile.

What I was going for in the chart was that these are all mobile stats but they fade out the further away from this year you get.

Feel free to ignore.

tunetheweb replied:
The fading does introduce colour contrast issues (a touchy subject as we've seen!)

I'm gonna ignore if you don't mind.

tunetheweb commented on src/content/en/2021/third-parties.md:
It was for the last two years.

rviscomi replied:
SGTM

tunetheweb commented on src/content/en/2021/third-parties.md:
Done across all charts and descriptions

tunetheweb commented on src/content/en/2021/third-parties.md:
Done.

tunetheweb commented on src/content/en/2021/third-parties.md:
An interesting bug in Google Sheets. If you mix label types (e.g. numbers and text like I had), then it shows fine in Sheets, but will not in the publish. You need to change them all to text (which screws up the order - so you have to add _another_ column with them as numbers to sort that). Anyway sorted now but one to keep an eye out for, for the other chapters. Tricky to figure out and get it to display properly.

tunetheweb commented on src/content/en/2021/third-parties.md:
Reworded instead to: "The high usage of HTML may seem surprising initially (surely documents would be the prevalent form of HTML and they would be first-party requests?), but our investigation showed them"

tunetheweb commented on src/content/en/2021/third-parties.md:
Fixed the subtitles (and also change the x-axis to note they are third-party requests).

I don't remember squashing it in, and playing around a bit more with the height I don't think it's any clearer even with more height. So think we should keep as is.

rviscomi replied:
SGTM

tunetheweb commented on src/content/en/2021/third-parties.md:
Fix subtitle and axis.

Same pushback against height changes - feel free to have a lot yourself and see what you think: https://docs.google.com/spreadsheets/d/1tf4RMF8SYr6he9tbqt61yuFJ_QK-F-i7XPxaPkpKSDI/edit#gid=1613966374

rviscomi replied:
I'll defer to you on the height thing

tunetheweb commented on src/content/en/2021/third-parties.md:
Done and done.

tunetheweb commented on src/content/en/2021/third-parties.md:
Done.

tunetheweb commented on src/content/en/2021/third-parties.md:
Accepted most of these except "top six" to "top 6". Our editors wiki says:

> Numbers less than ten should be written ("nine" not "9") except for percentages.

So shouldn't this stay as "six"?

rviscomi replied:
Usually yes, but there's another rule (not sure if in our wiki) that numbers should appear as digits for consistency with other numbers in the sentence. So "6" would be correct here, for the same reason why "8 of the top 15" isn't "eight of the top 15".

tunetheweb replied:
Fair point - changed.

tunetheweb commented on src/content/en/2021/third-parties.md:
Changed it to this, let me know if that is clearer:

> Using third parties can have a noticeable impact on performance. That's not necessarily a consequence of them being a third party per se. The same functionality implemented by a site owner as a first-party resource can often be less performant, given the expertise the third party should have on the particular field.

rviscomi replied:
LGTM!

tunetheweb commented on src/content/en/2021/third-parties.md:
Changed to:

```html
To avoid that hassle you can just use the hosted version, and ensure you're reducing the performance impact as much as possible, as discussed by <a hreflang="en" href="https://twitter.com/csswizardry">Harry Roberts</a> in his <a hreflang="en" href="https://csswizardry.com/2020/05/the-fastest-google-fonts/">The Fastest Google Fonts</a> post.
```

tunetheweb commented on src/content/en/2021/third-parties.md:
Reworded to:

> Twitter as a third-party can be used in one of two ways: as a retargeting advertising tracker, and as a way of embedding tweets. Embedding tweets in pages is more popular than other social networks. However it has been called out as having an undue impact on the page by many in the web performance community, including <a href="https://twitter.com/TheRealNooshu">Matt Hobbs</a> in his <a hreflang="en" href="https://nooshu.com/blog/2021/02/06/using-puppeteer-and-squoosh-to-fix-twitter-embeds/">Using Puppeteer and Squoosh to fix the web performance of embedded tweets</a> post. Our analysis backs that up—especially as those use cases will be diluted with the (presumably lighter) tracking use case in the above graph.

tunetheweb commented on src/content/en/2021/third-parties.md:
Accepted the suggestion, and clarified what I meant by "most":

> Similarly, image CDNs can optimize media better than most firsty-parties and, more importantly, can do this automatically without the need for manual steps that will inevitably be skipped or done incorrectly on occasion.

I think it's tough to make this more data driven since there are some factors at play (as I allude to in my first sentence). To really measure CDNs impact we'd need the same image on both.

I'm comfortable giving a brief breather from the stats here — as you'll have seen we're about to go very stat heavy and text-light with the embeds in this section anyway.

tunetheweb commented on src/content/en/2021/third-parties.md:
Fixed decimal and changed axis to "Percent of pages for which the third party is blocking"

tunetheweb commented on src/content/en/2021/third-parties.md:
I didn't like the apostrophe so changed it to "Main-thread blocking time of top 15 third parties."

I'd like to keep the order as consistent with the two graphs above.

tunetheweb commented on src/content/en/2021/third-parties.md:
Ooops missed this one. Added now.

tunetheweb commented on src/content/en/2021/third-parties.md:
Fixed

tunetheweb commented on src/content/en/2021/third-parties.md:
Changed it to this:

> We can see a huge impact of 1.6 seconds of main-thread activity at the median (50th percentile), rising to a shocking 4.6 seconds of main-thread blocking at the 90th percentile (still meaning 10% of websites have a worse impact than even that!). It should be remembered however that these are throttled, lab-simulated timings, so many real users may not be experiencing this level of impact, but it is still a lot.

tunetheweb commented on src/content/en/2021/third-parties.md:
Changed to:

> And finally, let's look at one further down the list: Twitter.

tunetheweb commented on src/content/en/2021/third-parties.md:
Changed it to "Google recommend's".

Gonna leave the irony comment in as still consider the post published by Google on web.dev. But good to know.

tunetheweb commented on src/content/en/2021/third-parties.md:
Done.

tunetheweb commented on src/content/en/2021/third-parties.md:
Changed to:

> Google Tag Manager fares much better than expected to be honest. This author has seen some horrific GTM implementations, overloaded with old tags and triggers that are no longer used. But GTM seems to do well at not blocking the main thread for too long in our test page loads.

tunetheweb commented:
Addressed most of the feedback @rviscomi . If you could have another look at the unresolved items?

I've also staged it here: https://20211105t024137-dot-webalmanac.uk.r.appspot.com/en/2021/third-parties

rviscomi commented on src/content/en/2021/third-parties.md:
@maudnals any feedback about this section?

rviscomi reviewed (approved):
LGTM @tunetheweb! Leaving one comment unresolved to give Maud a chance to review.

tunetheweb commented:
Merging this. We can include any further feedback if it comes in a separate PR.
